packages/did/src/DidDetails/FullDidDetails.ts

```diff
@@ -36,168 +34,146 @@ import {
   parseDidUri,
 } from '../Did.utils.js'
 
-import { DidDetails } from './DidDetails.js'
 import {
   getKeyRelationshipForExtrinsic,
   increaseNonce,
 } from './FullDidDetails.utils.js'
 
-export class FullDidDetails extends DidDetails {
-  public readonly identifier: DidIdentifier
-
-  /**
-   * Create an instance of [[FullDidDetails]] with the provided details.
-   * This is not to be used to create new DIDs, but it should only be used to serialize-deserialize full DID information to and from storage.
-   * Creating an instance of a full DID in this way without writing the information on the blockchain, will render the DID useless.
-   *
-   * @param creationDetails The creation details.
-   * @param creationDetails.identifier The DID subject identifier.
-   * @param creationDetails.uri The full DID URI.
-   * @param creationDetails.keys The set of public keys associated with the given full DID.
-   * @param creationDetails.keyRelationships The map of key ID -> relationship (e.g., authentication, attestation).
-   * @param creationDetails.serviceEndpoints The set of service endpoints controlled by the specified DID.
-   */
-  public constructor({
+/**
+ * Create a new instance of [[FullDidDetails]] after fetching the relevant information from the blockchain.
```

Comment:
If we proceed to remove the distinct types, I will update the file structure and the comments. When asking for a review I wasn’t confident you would approve of that change.

Reply:
Makes sense. What do you think of 'dummy types' along the lines of what I suggested in my main comment? This could be something like
```ts
interface LightDidDetails extends DidDetails {
  uri: `did:kilt:light:${string}`
}
```
and would primarily function to communicate which function works with what. In that case we could of course keep those functions in separate files.

Reply:
I think at the moment we do not have any function which only accepts light DIDs. We only have two functions which reject light DIDs. I feel that introducing top-level always visible classes just to communicate limitations of two functions is not only an overkill, but also creates more confusion than it clears up.

Reply:
could be overkill, yeah, especially if function names communicate constraints well (we talked about `parseDetailsFromLightDid` as a name for the one function I can think of that does only accept a light did).